ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/task/AbstractRetryingPeerTask.java

```diff
@@ -129,13 +130,12 @@ protected void handleTaskError(final Throwable error) {
           "No useful peer found, wait max 5 seconds for new peer to connect: current peers {}",
           ethContext.getEthPeers().peerCount());
 
-      final WaitForPeerTask waitTask = WaitForPeerTask.create(ethContext, metricsSystem);
       executeSubTask(
           () ->
               ethContext
-                  .getScheduler()
-                  // wait for a new peer for up to 5 seconds
-                  .timeout(waitTask, Duration.ofSeconds(5))
+                  .getEthPeers()
+                  .waitForPeer((peer) -> true)
```

Comment:
We do have the isSuitablePeer(peer) method on this class that we should use here.
```suggestion
                  .waitForPeer((peer) -> isSuitablePeer(peer))
```

Reply:
Yeah, that sounds good